src/java/org/apache/cassandra/service/paxos/PaxosRepair.java

```diff
@@ -659,45 +657,24 @@ static boolean validateVersionCompatibility(CassandraVersion version)
         return (version.major == 4 && version.minor > 0) || version.major > 4;
     }
 
-    static String getPeerVersion(InetAddressAndPort peer)
+    static boolean validatePeerCompatibility(IGossiper gossiper, Replica peer)
     {
-        EndpointState epState = Gossiper.instance.getEndpointStateForEndpoint(peer);
```

Comment:
this is duplicate logic from `Gossiper`, the only difference was

```
try
        {
            return value.value;
        }
        catch (IllegalArgumentException e)
        {
            return null;
        }
```

Not sure how a field access leads to `IllegalArgumentException`, so removed in favor of the gossip function